web/src/features/panels/zone/ZoneHeaderGauges.tsx

```diff
@@ -36,19 +36,16 @@ export function ZoneHeaderGauges({ data }: { data?: ZoneDetails }) {
   } = selectedData || {};
 
   const intensity = getCarbonIntensity(
-    isConsumption,
-    co2intensity,
-    co2intensityProduction
+    { c: { ci: co2intensity }, p: { ci: co2intensityProduction } },
+    isConsumption
   );
   const renewable = getRenewableRatio(
-    isConsumption,
-    renewableRatio,
-    renewableRatioProduction
+    { c: { rr: renewableRatio }, p: { rr: renewableRatioProduction } },
+    isConsumption
   );
   const fossilFuelPercentage = getFossilFuelRatio(
-    isConsumption,
-    fossilFuelRatio,
-    fossilFuelRatioProduction
+    { c: { fr: fossilFuelRatio }, p: { fr: fossilFuelRatioProduction } },
+    isConsumption
   );
```

Comment:
couldn't we just pass the whole "selectedData" variable to the util functions here instead of reconstructing the object?

Reply:
Possibly, and might be a better solution. I'll take a look!

Reply:
So this is because this is using the zone data and not the state data, when the state data is used it's passed like this:
```js
  const intensity = getCarbonIntensity(zoneData, isConsumption);
```

So if we want to use the same function for both one of them need reconstruct the object in one place or the other and I think it's better the zone details should do it  than the state related things as they are used more and need to be faster.

I'm considering changing the usage of the carbon intensity from the details file to the state files though as those are already loaded. That means they will be used the same way. But that is a separate PR for sure.

Reply:
It would be possible to make the function handle both kinds of data and have internal logic on which to use but unsure how beneficial that would be over constructing the object. So I'll leave that up to you to decide then I'll implement it if we decide to do so.

Reply:
Ah okay, I see - let's keep it like this then :)